text/src/autogluon/text/automm/predictor.py

```diff
@@ -786,30 +931,38 @@ def _fit(
             )
 
         if trainer.global_rank == 0:
-            self._top_k_average(
-                model=model,
-                save_path=save_path,
-                minmax_mode=minmax_mode,
-                is_distill=is_distill,
-                config=config,
-                val_df=val_df,
-                validation_metric_name=validation_metric_name,
-            )
+            # We do not perform averaging checkpoint in the case of hpo for each trial
+            # We only averaging the checkpoint of the best trial in the end in the master process
+            if not hpo_mode:
+                top_k_model_paths, last_ckpt_path, best_k_models_yaml_path = self._prepare_checkpoint(config, save_path, minmax_mode)
+
+                self._top_k_average(
+                    model=model,
+                    save_path=save_path,
+                    top_k_model_paths=top_k_model_paths,
+                    last_ckpt_path=last_ckpt_path,
+                    minmax_mode=minmax_mode,
+                    is_distill=is_distill,
+                    config=config,
+                    val_df=val_df,
+                    validation_metric_name=validation_metric_name,
+                )
+                
+                # remove the yaml file after cleaning the checkpoints
+                if os.path.isfile(best_k_models_yaml_path):
+                    os.remove(best_k_models_yaml_path)
         else:
             sys.exit(
                 f"Training finished, exit the process with global_rank={trainer.global_rank}..."
             )
-
-    def _top_k_average(
-            self,
-            model,
-            save_path,
-            minmax_mode,
-            is_distill,
-            config,
-            val_df,
-            validation_metric_name,
+            
+    def _prepare_checkpoint(
```

Comment:
Well, because this is what the regular training logic uses. HPO would use ray tune checkpoint, and the way to find `top_k_model_paths` is different. For example, there won't be `best_k_models.yaml` in hpo

Reply:
Makes sense